### PR TITLE
Allow SDS with a plaintext connection to pilot

### PR DESCRIPTION
**Pull request by howardjohn**

### CODE REVIEW

**Comment by howardjohn** [comment]
This needs more investigation, but from what I am seeing if the port is 15010 we are not configuring SDS which breaks the pod since there are no certs. I don't see why SDS has anything to do with control plane security

**Comment by nrjpoddar** [comment]
@howardjohn you already merged this in 1.5 afaict.

Did the tests break or report any failures?

**Comment by howardjohn** [comment]
@nrjpoddar I merged directly to 1.5 since it wasn't relevant to master except for this part. That one passed all 1.5 tests

File: pilot/cmd/pilot-agent/main.go [hunk 1]
```diff
@@ -435,7 +435,7 @@ var (
 
 			// If control plane auth is not mTLS or global SDS flag is turned off, unset UDS path and token path
 			// for control plane SDS.
-			if !controlPlaneAuthEnabled || !nodeAgentSDSEnabled {
+			if !nodeAgentSDSEnabled {
```
Comment (nrjpoddar) [inline pilot/cmd/pilot-agent/main.go]: Update the comment above your change as it's misleading.

I think control plane auth is needed to do SDS else you will be transporting certs over plain-text.

Comment (howardjohn) [reply pilot/cmd/pilot-agent/main.go]: Its a bit orthogonal, isn't it? SDS is now the only support mechanism to serve certs to Envoy, there is no file mounts. So we should either get rid of the plaintext port entirely or make it work?

Comment (nrjpoddar) [reply pilot/cmd/pilot-agent/main.go]: I'm with you on that. Basically the summary is:

* If SDS is enabled, control plane security must be turned on and non-secure ports blocked.
* If control plane security is disabled for whatever reason, I want to not use SDS and if file mounts are not configured, I would rather have traffic not working vs leaking certs.

Comment (howardjohn) [reply pilot/cmd/pilot-agent/main.go]: Ok then lets just kill the plaintext port then

Comment (nrjpoddar) [reply pilot/cmd/pilot-agent/main.go]: Works for me. What are the implications for users:

* Does any istioctl command relies on reaching Pilot via unsecure port?

Comment (howardjohn) [reply pilot/cmd/pilot-agent/main.go]: Its just XDS, istioctl connects over a different unsecure port 8080

cc @costinm

**Pull request closed without merging**